include/deal.II/base/exceptions.h

```diff
@@ -1659,11 +1659,20 @@ namespace deal_II_exceptions
 #    endif /*ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST*/
 #  endif   /*KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST*/
 #else      /*ifdef DEBUG*/
-#  define Assert(cond, exc) \
-    do                      \
-      {                     \
-      }                     \
+/*
+ * In order to avoid unused parameters (etc.) warnings we need to use cond
+ * and exc without actually evaluating the expression and generating code.
+ * We accomplish this by using decltype(...) and create a dummy function
+ * pointer with these signatures.
+ */
+#  define Assert(cond, exc)                           \
+    do                                                \
+      {                                               \
+        void (*a)(decltype(cond) *, decltype(exc) *); \
+        (void)a;                                      \
```

Comment:
```suggestion
        void (*assert_fcn_ptr)(decltype(cond) *, decltype(exc) *); \
        (void)assert_fcn_ptr;                                      \
```
I'm being picky, but we should avoid using `a` here in case someone is using `-Wshadow` (and has a local variable named `a`).

Reply:
fixed.